README.md
# solydk-system-adjustments
System adjustments for SolydK


debian/changelog
solydk-system-adjustments (2.5.8) solydxk; urgency=low

  * Replace LibreOffice start menu favorite

 -- Arjen Balfoort <arjenbalfoort@solydxk.com>  Mon, 06 Jun 2016 16:14:54 +0200

solydk-system-adjustments (2.5.7) solydxk; urgency=low

  * Updated mimeapps.list

 -- Arjen Balfoort <arjenbalfoort@solydxk.com>  Thu, 02 Jun 2016 10:56:46 +0200

solydk-system-adjustments (2.5.6) solydxk; urgency=low

  * Added /etc/xdg/kglobalshortcutsrc to configure win key opening start menu

 -- Arjen Balfoort <arjenbalfoort@solydxk.com>  Fri, 15 Apr 2016 14:02:04 +0200

solydk-system-adjustments (2.5.5) solydxk; urgency=low

  * Removed trigger on solydxk-system service

 -- Arjen Balfoort <arjenbalfoort@solydxk.com>  Sat, 19 Mar 2016 10:30:23 +0100

solydk-system-adjustments (2.5.4) solydxk; urgency=low

  * Moved /etc/skel/.gnupg to solydxk-system package

 -- Arjen Balfoort <arjenbalfoort@solydxk.com>  Sat, 27 Feb 2016 16:00:00 +0100

solydk-system-adjustments (2.5.3) solydxk; urgency=low

  * Add sha512sum to Service Menu

 -- Arjen Balfoort <arjenbalfoort@solydxk.com>  Mon, 22 Jan 2016 10:00:00 +0100

solydk-system-adjustments (2.5.2) solydxk; urgency=low

  * Added new default flat design
  * Changed desktop layout to folderview

 -- Arjen Balfoort <arjenbalfoort@solydxk.com>  Fri, 29 Jan 2016 18:00:00 +0100

solydk-system-adjustments (2.2.0) solydxk; urgency=low

  * Updated mimeapps.list

 -- Arjen Balfoort <arjenbalfoort@solydxk.com>  Mon, 18 Jan 2016 19:00:00 +0100

solydk-system-adjustments (2.1.9) solydxk; urgency=low

  * Updated for EE

 -- Arjen Balfoort <arjenbalfoort@solydxk.com>  Sun, 29 Nov 2015 13:05:34 +0100

solydk-system-adjustments (2.1.8) solydxk; urgency=low

  * Fix for SolydK BO users

 -- Arjen Balfoort <arjenbalfoort@solydxk.com>  Thu, 22 Oct 2015 21:00:00 +0100

solydk-system-adjustments (2.1.7) solydxk; urgency=low

  * Added female user picture

 -- Arjen Balfoort <arjenbalfoort@solydxk.com>  Wed, 07 Oct 2015 12:00:00 +0100

solydk-system-adjustments (2.1.6) solydxk; urgency=low

  * Resolved theme bug for PlayOnLinux (and probably others) to switch Gtk theme to oxygen-gtk

 -- Arjen Balfoort <arjenbalfoort@solydxk.com>  Thu, 10 Sep 2015 10:00:00 +0100

solydk-system-adjustments (2.1.5) solydxk; urgency=low

  * Added fancy progressbar with ANSI 256 color codes for terminal

 -- Arjen Balfoort <arjenbalfoort@solydxk.com>  Wed, 19 Aug 2015 17:00:00 +0100

solydk-system-adjustments (2.1.1) solydxk; urgency=low

  * Resolved issue inserting duplicate keys in LightDM configuration

 -- Arjen Balfoort <arjenbalfoort@solydxk.com>  Sun, 17 May 2015 14:00:00 +0100

solydk-system-adjustments (2.1.0) solydxk; urgency=low

  * Changed LightDM configuration for Debian testing

 -- Arjen Balfoort <arjenbalfoort@solydxk.com>  Fri, 15 May 2015 11:00:00 +0100

solydk-system-adjustments (2.0.9) solydxk; urgency=low

  * Divert Lines wallpapers

 -- Arjen Balfoort <arjenbalfoort@solydxk.com>  Mon, 27 Apr 2015 10:00:00 +0100

solydk-system-adjustments (2.0.8) solydxk; urgency=low

  * Make baloo a recommended package, not a dependency

 -- Arjen Balfoort <arjenbalfoort@solydxk.com>  Tue, 17 Mar 2015 18:00:00 +0100

solydk-system-adjustments (2.0.7) solydxk; urgency=low

  * Do not update-grub for armhf
  * Added dependencies gtk2-engines-pixbuf, oxygencursors

 -- Arjen Balfoort <arjenbalfoort@solydxk.com>  Fri, 13 Mar 2015 10:00:00 +0100

solydk-system-adjustments (2.0.6) solydxk; urgency=low

  * Added resolutions for ksplash theme

 -- Arjen Balfoort <arjenbalfoort@solydxk.com>  Sat, 07 Mar 2015 23:00:00 +0100

solydk-system-adjustments (2.0.5) solydxk; urgency=low

  * Moved python-apt templates to solydxk-system

 -- Arjen Balfoort <arjenbalfoort@solydxk.com>  Thu, 05 Mar 2015 11:00:00 +0100

solydk-system-adjustments (2.0.4) solydxk; urgency=low

  * Changed evolvere dependency

 -- Arjen Balfoort <arjenbalfoort@solydxk.com>  Wed, 18 Feb 2015 19:00:00 +0100

solydk-system-adjustments (2.0.2) solydxk; urgency=low

  * greybird-solydk-gtk3 theme updated

 -- Arjen Balfoort <arjenbalfoort@solydxk.com>  Sat, 07 Feb 2015 11:00:00 +0100

solydk-system-adjustments (2.0.0) solydxk; urgency=low

  * Merged with: plymouth-themes-solydk, solydk-artwork, solydk-grub2-theme, solydk-lightdm-adjustments
  * Changed to QtCurve instead of Oxygen theme
  * Changed to Evolvere icon theme

 -- Arjen Balfoort <arjenbalfoort@solydxk.com>  Sun, 14 Sep 2014 17:00:00 +0100

solydk-system-adjustments (1.3.3) solydxk; urgency=low

  * Automount detachable USB devices

 -- Arjen Balfoort <arjenbalfoort@solydxk.com>  Thu, 28 Aug 2014 09:00:00 +0100

solydk-system-adjustments (1.3.2) solydxk; urgency=low

  * Removed skel files for chromium

 -- Arjen Balfoort <arjenbalfoort@solydxk.com>  Mon, 04 Aug 2014 21:00:00 +0100

solydk-system-adjustments (1.3.1) solydxk; urgency=low

  * Changed Kickoff menu favorites list
  * Added password-less printer configuration and mounts

 -- Arjen Balfoort <arjenbalfoort@solydxk.com>  Mon, 07 Jul 2014 13:00:00 +0100

solydk-system-adjustments (1.3.0) solydxk; urgency=low

  * Removed startkde: now part of kde-workspace-bin

 -- Arjen Balfoort <arjenbalfoort@solydxk.com>  Fri, 20 Jun 2014 14:00:00 +0100

solydk-system-adjustments (1.2.9) solydxk; urgency=low

  * Default font to Clear Sans
  * Xhost security issue resolved
  * Divert non-solyd wallpapers

 -- Arjen Balfoort <arjenbalfoort@solydxk.com>  Thu, 17 Apr 2014 12:00:00 +0100

solydk-system-adjustments (1.2.5) solydxk; urgency=low

  * Removed Firefox settings
  * Restart pulseaudio after sleep

 -- Arjen Balfoort <arjenbalfoort@solydxk.com>  Fri, 07 Feb 2014 18:00:00 +0100

solydk-system-adjustments (1.2.0) solydxk; urgency=low

  * Removed Thunderbird settings

 -- Arjen Balfoort <arjenbalfoort@solydxk.com>  Thu, 06 Feb 2014 14:00:00 +0100

solydk-system-adjustments (1.1.9) solydxk; urgency=low

  * Changed default font from Ubuntu to Arial
  * Fix kgpg init


usr/share/solydxk/adjustments/plasma_shells_org.kde.plasma.desktop_contents_layout.js
loadTemplate("org.kde.plasma.desktop.defaultPanel")

for (var i = 0; i < screenCount; ++i) {
    //var id = createActivity("Desktop");
    var id = createActivity("Desktop", "org.kde.plasma.folder");
    var desktopsArray = desktopsForActivity(id);
    print(desktopsArray.length);
    for( var j = 0; j < desktopsArray.length; j++) {
        desktopsArray[j].currentConfigGroup = ["Wallpaper","General"];
        desktopsArray[j].writeConfig("Color", "142,182,201");
        desktopsArray[j].wallpaperPlugin = 'org.kde.image';
        desktopsArray[j].currentConfigGroup = ["Wallpaper","org.kde.image","General"];
        desktopsArray[j].writeConfig("FillMode", 2);
        desktopsArray[j].writeConfig("Color", "142,182,201");
        //desktopsArray[j].addWidget("org.kde.plasma.folder");
    }
}


usr/share/solydxk/adjustments/plasma_layout-templates_org.kde.plasma.desktop.defaultPanel_contents_layout.js
var applaunchericon = "solydk-flat"

var filemanagericon = "file-manager"
var filemanagername = "dolphin"
var filemanagerdesc = "File Manager"
var filemanagerurl = "/usr/share/applications/org.kde.dolphin.desktop"

var panel = new Panel
panel.location = "bottom"
panel.height = 36

var applauncher = panel.addWidget("org.kde.plasma.kickoff")
applauncher.writeConfig("icon", applaunchericon)
applauncher.currentConfigGroup = ["Shortcuts"]
applauncher.writeConfig("global", "Alt+F1")

panel.addWidget("org.kde.plasma.showdesktop")

var fmlauncher = panel.addWidget("org.kde.plasma.icon")
fmlauncher.writeConfig("iconName", filemanagericon)
fmlauncher.writeConfig("applicationName", filemanagername)
fmlauncher.writeConfig("genericName", filemanagerdesc)
fmlauncher.writeConfig("url", filemanagerurl)

panel.addWidget("org.kde.plasma.pager")
panel.addWidget("org.kde.plasma.taskmanager")
panel.addWidget("org.kde.plasma.systemtray")
panel.addWidget("org.kde.plasma.digitalclock")


usr/share/solydxk/default-settings/kde4-profile/default/share/apps/plasma-desktop/updates/02-solydk-menu-icon.js
activity.currentConfigGroup = new Array('ToolBox')
activity.writeConfig('corner', '1')
activity.writeConfig('offset', '0')

for (var i = 0; i < panelIds.length; ++i) {
    var panel = panelById(panelIds[i]);
    var widgetIds = panel.widgetIds;

    for (var j = 0; j < widgetIds.length; ++j) {
        var widget = panel.widgetById(widgetIds[j]);

        if (widget && (widget.type == 'quickaccess') &&
            (widget.readConfig("icon", "") != "user-home")) {
                widget.writeConfig("icon", "user-home");
        } // if quickaccess

        if (widget && (widget.type == 'lancelot_launcher') &&
            (widget.readConfig("icon", "") != "solydk-flat")) {
                widget.writeConfig("icon", "solydk-flat");
        } // if lancelot_launcher


        if (widget && (widget.type == 'launcher') &&
            (widget.readConfig("icon", "") != "solydk-flat")) {
                widget.writeConfig("icon", "solydk-flat");
        } // if launcher
    } // for widgets in panel
} // for panel




usr/share/solydxk/default-settings/kde4-profile/default/share/apps/plasma-desktop/init/00-defaultLayout.js
// Placed in /usr/share/solydxk/default-settings/kde4-profile/default/share/apps/plasma-desktop/init/
// This script is run for new users, which do not have a .kde directory
// and it set's the default wallpaper for all activities

// Wallpaper theme: sub directory of /usr/share/wallpapers/
var wallpaper = "solydk-flat"
var launchericon = "solydk-flat"
var dolphindesktop = "/usr/share/applications/kde4/dolphin.desktop"
var backgroundrgb = "142,182,201"

for (var i = 0; i < screenCount; ++i) {
  //desktop = new Activity
  var desktop = new Activity("folderview")
  desktop.name = i18n("Desktop")
  desktop.screen = i
  desktop.wallpaperPlugin = 'image'
  desktop.wallpaperMode = 'SingleImage'
  desktop.currentConfigGroup = Array('Wallpaper', 'image')
  desktop.writeConfig('wallpaper', wallpaper)
  desktop.writeConfig("userswallpaper", wallpaper)
  desktop.writeConfig('wallpaperposition', '2')          //2 enables croping
  desktop.writeConfig('wallpapercolor', backgroundrgb)
  desktop.currentConfigGroup = Array('Wallpaper', 'color')
  desktop.writeConfig('color1', backgroundrgb)
  desktop.currentConfigGroup = new Array('ToolBox')
  desktop.writeConfig('corner', '1')
  desktop.writeConfig('offset', '0')
  
  // Add desktop folderview if no folderviews exist
  //var folderview
  //if (desktop.widgets("folderview").length == 0) {
  //  folderview = desktop.addWidget("folderview")
  //} else {
  //  folderview = desktop.widgets("folderview")
  //}
  //folderview.writeConfig("url", "desktop:/")
  //folderview.height = 200
  //folderview.width = 300
  //folderview.x = screenGeometry(0).width / 2

  panel = new Panel
  panel.screen = i
  panel.location = 'bottom'
  //panel.height = panels()[i].height = screenGeometry(0).height > 1024 ? 35 : 27
  panel.height = 36

  // Add SolydXK widgets
  launcher = panel.addWidget("launcher")
  launcher.globalShortcut = "F13"
  launcher.writeConfig("icon", launchericon)

  panel.addWidget("showdesktop")
        
  var widget = panel.addWidget("icon")
  widget.writeConfig("Url", dolphindesktop)
        
  var tasks = panel.addWidget("tasks")
  tasks.writeConfig("showOnlyCurrentScreen", true);

  var systray = panel.addWidget("systemtray")
  j = 0;
  if (hasBattery) {
    systray.currentConfigGroup = new Array("Applets", ++j)
    systray.writeConfig("plugin", "battery")
  }
  systray.currentConfigGroup = new Array("Applets", ++j)
  systray.writeConfig("plugin", "message-indicator")
  systray.currentConfigGroup = new Array("Applets", ++j)
  systray.writeConfig("plugin", "org.kde.networkmanagement")
  systray.currentConfigGroup = new Array("Applets", ++j)
  systray.writeConfig("plugin", "notifier")

  var clock = panel.addWidget("digital-clock")
  clock.writeConfig("showDate", "false")
  
}
